Xamarin.Forms.Platform.iOS/Renderers/ShellTableViewController.cs

```diff
@@ -91,5 +102,26 @@ void OnScrolled(object sender, UIScrollView e)
 
 			LayoutParallax();
 		}
+
+		float SafeAreaOffset
+		{
+			get
+			{
+				if (!Forms.IsiOS11OrNewer)
+				{
+
+					return (float)UIApplication.SharedApplication.StatusBarFrame.Size.Height;
+
+				}
+
+				if (UIApplication.SharedApplication.KeyWindow != null)
+					return (float)UIApplication.SharedApplication.KeyWindow.SafeAreaInsets.Top;
+
+				if (UIApplication.SharedApplication.Windows.Length > 0)
+					return (float)UIApplication.SharedApplication.Windows[0].SafeAreaInsets.Top;
```

Comment:
What does iOS11+ require? `KeyWindow` and `Windows` require iOS 2.0+

Reply:
The problem here is timing. I found that the KeyWindow doesn't get set at the point in time where I need this value so the second check is just a fallback if the KeyWindow hasn't been set yet.  I moved this code into Platform so it can at least be centralized if there are other cases that need it which will mainly just be components that sit above everything and need to know window constraints